src/v/serde/test/bench.cc

```diff
@@ -88,3 +93,100 @@ PERF_TEST(big_10mb, serialize) {
 PERF_TEST(big_10mb, deserialize) {
     return deserialize_big(10 << 20 /*10MB*/, 1 << 15 /*32KB*/);
 }
+
+cluster::topic_status make_topic_status(size_t id, size_t num_partitions) {
+    cluster::topic_status ts;
+    ts.tp_ns = model::topic_namespace(
+      model::ns("foo"), model::topic("bar" + std::to_string(id)));
+
+    for (size_t i = 0; i < num_partitions; ++i) {
+        cluster::partition_status part_status;
+        part_status.id = model::partition_id(i);
+        part_status.leader_id = model::node_id(1);
+        part_status.reclaimable_size_bytes = 100;
+        part_status.revision_id = model::revision_id(1);
+        part_status.term = model::term_id(1);
+
+        ts.partitions.push_back(part_status);
+    }
+
+    return ts;
+}
+
+cluster::node_health_report
+make_node_health_report(size_t num_topics, size_t partitions_per_topic) {
+    cluster::node_health_report hr;
+    hr.id = model::node_id(1);
+
+    hr.local_state.redpanda_version = cluster::node::application_version(
+      "v21.1.1");
+    hr.local_state.logical_version = cluster::cluster_version(1);
+    hr.local_state.uptime = std::chrono::milliseconds(100);
+
+    hr.local_state.data_disk.path = "/bar/baz/foo/foo/foo/foo/foo/foo/foo/bar";
+    hr.local_state.data_disk.total = 1000;
+    hr.local_state.data_disk.free = 500;
+
+    for (size_t i = 0; i < num_topics; ++i) {
+        hr.topics.push_back(make_topic_status(i, partitions_per_topic));
+    }
+
+    return hr;
+}
+
+__attribute__((noinline)) void do_bench_serialize_node_health_report(
```

Comment:
note: I think `[[gnu::noinline]]` is preferred today

Reply:
fixed